gobblin-runtime/src/main/java/org/apache/gobblin/runtime/task/TaskIFaceWrapper.java

```diff
@@ -54,6 +53,7 @@ public TaskIFaceWrapper(TaskIFace underlyingTask, TaskContext taskContext, Count
     this.taskId = taskContext.getTaskState().getTaskId();
     this.countDownLatch = countDownLatch;
```

Comment:
Have you figured out why the NoArgsConstructor is enforced and you cannot pass those values through the constructor? This type of implementation seems weird.